src/scheduler/MpiWorld.cpp

```diff
@@ -222,20 +222,31 @@ void MpiWorld::create(faabric::Message& call, int newId, int newSize)
         msg.set_mpiworldid(id);
         msg.set_mpirank(i + 1);
         msg.set_mpiworldsize(size);
-        // Log chained functions to generate execution graphs
-        if (thisRankMsg != nullptr && thisRankMsg->recordexecgraph()) {
-            sch.logChainedFunction(call.id(), msg.id());
-            msg.set_recordexecgraph(true);
+        if (thisRankMsg != nullptr) {
+            // Set message fields to allow for function migration
+            msg.set_appid(thisRankMsg->appid());
+            msg.set_cmdline(thisRankMsg->cmdline());
+            msg.set_inputdata(thisRankMsg->inputdata());
+            msg.set_migrationcheckperiod(thisRankMsg->migrationcheckperiod());
+            // Log chained functions to generate execution graphs
+            if (thisRankMsg->recordexecgraph()) {
+                sch.logChainedFunction(call.id(), msg.id());
+                msg.set_recordexecgraph(true);
+            }
         }
     }
 
     std::vector<std::string> executedAt;
     if (size > 1) {
-        // Send the init messages (note that message i corresponds to rank i+1)
-        // By default, we use the NEVER_ALONE policy for MPI executions to
-        // minimise cross-host messaging
+        // 10/01/22 - We add this check to force different scheduling behaviour
+        // if running migration experiments. The check can be deleted
+        // afterwards.
```

Comment:
Can we remove this now and have a way to programmatically do it? We could add a `topologyHint` parameter on the message itself alongside the `migrationCheckPeriod`. The `Message` is getting very bloated, but I've created an issue around refactoring it: https://github.com/faasm/faabric/issues/221